pkg/api/entries.go

```diff
@@ -287,12 +294,33 @@ func createLogEntry(params entries.CreateLogEntryParams) (models.LogEntry, middl
 		return nil, handleRekorAPIError(params, http.StatusBadRequest, err, fmt.Sprintf(validationError, err))
 	}
 
-	areEntryAlgorithmsAllowed, err := checkEntryAlgorithms(entry)
+	checkedAlgorithmResult, err := checkEntryAlgorithms(entry)
 	if err != nil {
 		return nil, handleRekorAPIError(params, http.StatusBadRequest, err, fmt.Sprintf(validationError, err))
 	}
-	if !areEntryAlgorithmsAllowed {
-		return nil, handleRekorAPIError(params, http.StatusBadRequest, errors.New("entry algorithms are not allowed"), fmt.Sprintf(validationError, "entry algorithms are not allowed"))
+	if !checkedAlgorithmResult.allowed {
+		var publicKeyInfo string
+		switch pKey := checkedAlgorithmResult.publicKey.(type) {
+		case *ecdsa.PublicKey:
+			switch pKey.Curve {
+			case elliptic.P256():
+				publicKeyInfo = "ecdsa public key: P256"
+			case elliptic.P384():
+				publicKeyInfo = "ecdsa public key: P384"
+			case elliptic.P521():
+				publicKeyInfo = "ecdsa public key: P521"
+			default:
+				publicKeyInfo = "ecdsa public key"
+			}
+		case *rsa.PublicKey:
+			publicKeyInfo = fmt.Sprintf("rsa public key: %T/%v", pKey, pKey.Size())
+		case *ed25519.PublicKey:
+			publicKeyInfo = fmt.Sprintf("ed25519 public key: %T", pKey)
+		default:
+			publicKeyInfo = fmt.Sprintf("public key: %T", pKey)
+		}
+		algorithmInfo := fmt.Sprintf("entry algorithm %v/%v not allowed", publicKeyInfo, checkedAlgorithmResult.hash)
+		return nil, handleRekorAPIError(params, http.StatusBadRequest, errors.New("entry algorithms are not allowed"), fmt.Sprintf(validationError, algorithmInfo))
```

Comment:
The last two params should be flipped, the error logged is the second to last param, the error message for the client is the last param.

If you want a more descriptive error message, we'll need to update [the e2e test](https://github.com/sigstore/rekor/blob/30c2d4e11c480402d93258b463d2ae03a2029ad1/tests/client-algos-e2e-test.sh#L171) too